### ISSUE TITLE
Local settings file for sensitive data

### ISSUE THREAD

**Issue by rachelwhitton** [pantheon-systems/documentation#1129]
Surfaced in Power Users thread **Local settings file for sensitive data**

### Document the following example provided by Josh:

Your only options for storing durable state data are the database, git, or the shared filesystem.

The alternative Steve suggested using the filesystem would be to place the secrets in a json or ini-style text file kept in `/sites/default/files/private`, which you can "deploy" with SFTP, and then read from `settings.php`. 

If you name the file something distinctive `stripe_live.json` you could easily have code like this in `settings.php`:

```
<?php

if (isset($_ENV['PANTHEON_ENVIRONMENT']) && $_ENV['PANTHEON_ENVIRONMENT'] == 'live') {
  $json_text = file_get_contents('sites/default/files/private/stripe_live.json');
  $stripe_data = json_load($json_text, 1);
  $conf['stripe_key'] = $stripe_data['key'];
}
else {
  // We aren't in prod, load a fallback or null key.
  $conf['stripe_key'] = 'foo';
}
```
This would result in the key file being distributed to other environments, but it would not be in the repository. If you are worried about other people seeing the file in development, then this is no better than the database. However, if your concern is dev environments accidentally making calls to the production stripe account, this could work for you.

### Follow up note from user on D8:
While this was really helpful, looks like ye ole `$conf` array is no longer around in D8. Instead, we need to do something like `\Drupal::config->set()`, which throws an error when I try to use it in the settings.php context. I'll have to do some more research on per-environment configuration management.

The example you gave is very useful for D6/D7 sites though & worth adding to the Pantheon docs IMO

**Comment by nataliejeremy** [comment]
@rachel do you have a specific doc in mind?

**Comment by rachelwhitton** [comment]
@natalie, the content is Drupal specific, though the same logic applies to WP sites. Looking over it again, maybe it's a good candidate for a new doc? Something like `docs/articles/sites/storing-sensitive-data.md` The suggestioned PHP code should work for both Drupal and WP, though CSE should confirm. I can draft a new doc once we triage/prioritize the issue

**Comment by bmackinney** [comment]
There are good code examples in https://github.com/pantheon-systems/quicksilver-examples/.

**Comment by rachelwhitton** [comment]
Closed by #1445
